internal/util/k8s.go

```diff
@@ -448,7 +449,22 @@ func CreateChaincodePod(
 		chaincodeData,
 	)
 
-	err := deleteChaincodePod(ctx, logger, podsClient, podName, namespace, chaincodeData)
+	peerPod, err := podsClient.Get(ctx, peerID, metav1.GetOptions{})
```

Comment:
This doesn't look like the right way to get the peer pod. Instead, I think there needs to be something to look up the pod that the builder is running on, although I'm not sure how to do that off the top of my head.

Reply:
The idea here was that whatever namespace is being passed to the ```Create Chain code pod``` will be the only namespace from which (if a peer pod exist ) can be added as an owner and the podsClient passed is created from the same namespace.

Reply:
That makes sense, however I don't think there's currently any way to identify the peer pod. The builder would be running in the same container as the peer, so it would be in the same pod, but I've not found a way to get the current pod using the kubernetes client API (I'm probably just missing something obvious). The [Downward API](https://kubernetes.io/docs/concepts/workloads/pods/downward-api/) would work but that would add some requirements on the way the peer pod is defined, plus probably more config for the k8s builder.